### ISSUE TITLE
Pattern Validation for enums. At the moment Krasa supports only String type for Patterns

### ISSUE THREAD

**Issue by rohanmukesh** [krasa/krasa-jaxb-tools#48]
At the moment Krasa only supports pattern for Strings.
We have a business scenario where we have Enum which consists of 10 values and There are many complex types which refer to enum but then we need it to be restricted to use only 2 enum values or 3 instead of all of them.

Here is a business case for this ::
ActionEnum:

```
 <xsd:simpleType name="ActionValues">
        <xsd:restriction base="xsd:string">
            <xsd:enumeration value="Flights"/>
            <xsd:enumeration value="Delta"/>
            <xsd:enumeration value="FareValue"/>
 </xsd:restriction>
    </xsd:simpleType>
```

complex type:

```
<xsd:complexType name="FlightsDeltaAction">
        <xsd:sequence>
            <xsd:element name="name">
                <xsd:simpleType>
                    <xsd:restriction base="ActionValues">
                        <xsd:pattern value="Flights"/>
                        <xsd:pattern value ="Delta"/>
                    </xsd:restriction>
                </xsd:simpleType>
```

I can see from the sourcecode that there is below code:
else if (simpleType.getFacet("pattern") != null) {
            String pattern = simpleType.getFacet("pattern").getValue().value;
            if ("String".equals(field.type().name())
 Can we have this extended to have restrictions for Enums as well?

Thanks,
Rohan

**Comment by krasa** [comment]
Pull request welcomed

**Comment by krasa** [comment]
I think it was fixed by #60 and #69
